blog/2023-09-15-wing-github-action.md

```diff
@@ -0,0 +1,71 @@
+authors: 
+  - skorfmann
+tags: [winglang, github actions, ci/cd]
+hide_table_of_contents: true
+---
+
+> Unlocking CI/CD with the Wing Github Action! 
+> <!--truncate-->
+
+Winglang is a specialized programming language designed for cloud applications. It smoothly integrates infrastructure and application code, enabling developers to optimize their workflow. This language translates into Infrastructure as Code (IaC) configurations, such as Terraform or CloudFormation, for cloud resources, and utilizes JavaScript for executing business logic. Importantly, neither Winglang nor its Command Line Interface (CLI) assume how users deploy applications using Terraform. For those utilizing GitHub, we've developed a dedicated Winglang GitHub Action.
+
+GitHub Actions is a renowned automation platform within GitHub, facilitating developers in tailoring, executing, and automating their software development workflows directly within their repositories. From building and deploying applications to automating routine tasks like responding to issues, GitHub Actions enhances continuous integration and continuous deployment (CI/CD) by initiating workflows in reaction to specific events, including push and pull requests or defined schedules.
+
+While developers have the autonomy to create a custom workflow integrating Winglang with GitHub Actions, the Wing GitHub Action is designed to be comprehensive and simplify the Wing and Terraform setup process. Our objective is to enable rapid deployment and introduce standardized deployment workflows.
+
+### Wing Action
+
+The [Wing Github Action](https://github.com/winglang/wing-github-action) enables two use-cases at the moment:
+
+#### 1. Deployments
+
+To begin, here is the basic configuration for deployments:
+
+```yml
+steps:
+  - name: Deploy Winglang App
+    uses: winglang/wing-github-action/actions/deploy@0.1.0
+    with:
+      entry: 'main.wing' # Required, replace this with your entry file if different
+      target: 'tf-aws' # Required, the target to deploy to. e.g. tf-aws, tf-gcp, tf-azure
+    env:
+      TF_BACKEND_BUCKET: '<your-bucket-name>' # required
+      TF_BACKEND_BUCKET_REGION: '<your-bucket-region>' #  required
+```
+
+This does exactly what you think it might do. It compiles the Wing application to Terraform and Javascript to eventually run a Terraform apply. Right now, a S3 bucket is the only [Terraform backend](https://www.winglang.io/docs/guides/terraform-backends) which is supported - however, there's an open issue to allow defining your own via Wing compiler plugins. Other than that, the action is fully self contained and includes all neccessary dependencies.
+
+#### 2. Pull Request Diffs
+
+The configuration for pull request diffs closely resembles the deployment setup, with the primary distinction being the `github-token``, necessary for posting comments on pull requests:
+
+```yml
+steps:
+  - name: Pull Request Diff
+    uses: winglang/wing-github-action/actions/pull-request-diff@main
```

Comment:
Using the version from @main? Or do you want to link to a specific release, like you're doing on the deployment action?